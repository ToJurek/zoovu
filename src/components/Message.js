import React from "react";
import "../styles/Message.css"

export default class Message extends React.Component{

    render() {
        setTimeout(() => {
            window.location.reload(false);
        }, 10000);
        return (
            <div className={"message"}>
                Congratulations {this.props.name}. You're time: {this.props.time} seconds.
            </div>
        )
    }
}