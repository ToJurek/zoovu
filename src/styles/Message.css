.message {
    font-size: 30px;
    padding-top: 50px;
}