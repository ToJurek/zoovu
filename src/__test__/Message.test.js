import React from "react"
import Enzyme, {shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Message from "../components/Message";
import renderer from 'react-test-renderer'
var sinon = require('sinon');

Enzyme.configure({adapter: new Adapter()})

describe("Message Component", () => {
    it('should show text', function () {
        const wrapper = shallow(<Message name={"user"} time={10}/>)
        const text = wrapper.find("div")
        expect(text.text()).toBe("Congratulations user. You're time: 10 seconds.")
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(<Message name={"user"} time={10}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    xit('should call ', function () {
        var clock = sinon.useFakeTimers();
        const wrapper = shallow(<Message name={"user"} time={10}/>)
        const reload = jest.spyOn(wrapper.instance(), 'window.location.reload()');
        clock.tick(9999)
        wrapper.instance().reload();
        expect(reload).toHaveBeenCalledTimes(0);
        clock.tick(10001)
        wrapper.instance().reload();
        expect(reload).toHaveBeenCalledTimes(1);
    });


})

