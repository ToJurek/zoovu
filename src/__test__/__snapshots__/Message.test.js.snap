// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component renders correctly 1`] = `
<div
  className="message"
>
  Congratulations 
  user
  . You're time: 
  10
   seconds.
</div>
`;
